src/app/(components)/top-loader.tsx

```diff
@@ -0,0 +1,24 @@
+"use client";
+import * as React from "react";
+import * as NProgress from "nprogress";
+import NextTopLoader from "nextjs-toploader";
+import { usePathname, useRouter, useSearchParams } from "next/navigation";
+
+export function TopLoader() {
+  return (
+    <>
+      <NextTopLoader showSpinner={false} />
+      <FinishingLoader />
+    </>
+  );
+}
+
+function FinishingLoader() {
+  const pathname = usePathname();
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  React.useEffect(() => {
+    NProgress.done();
+  }, [pathname, router, searchParams]);
+  return null;
```

Comment:
Works, but is there any reason why we are separating this to a separate component? Can we can add the effect directly on the TopLoader instead?

Reply:
Just by precaution, I don't know about how the `<NextTopLoader>` component is implemented but i don't want route changes to cause unnecessary rerenders or effects rerunning of the `<NextTopLoader>` component.

Reply:
unrelated but can you tell me what `(components)` in `src/app/(components)/top-loader.tsx` means?

I can't find documentation about this.

Reply:
`()` denotes route groups which are just paths that do not appear in the URL bar.

The docs : https://nextjs.org/docs/app/building-your-application/routing/route-groups

Reply:
@Fredkiss3 Thanks so much!

Reply:
In my case will show conflict warnings after build

```js
 ⚠ Entire page /auth-demo/classic/login deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth-demo/classic/login
 ⚠ Entire page /auth-demo/modern/forgot-password deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth-demo/modern/forgot-password
 ⚠ Entire page /auth-demo/classic/new-password deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth-demo/classic/new-password
 ⚠ Entire page /404 deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /404
 ⚠ Entire page /auth/amplify/register deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/amplify/register
 ⚠ Entire page /auth/amplify/new-password deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/amplify/new-password
 ⚠ Entire page /auth-demo/modern/register deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth-demo/modern/register
 ⚠ Entire page /auth-demo/modern/verify deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth-demo/modern/verify
 ⚠ Entire page /about-us deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /about-us
 ⚠ Entire page /auth/auth0/callback deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/auth0/callback
 ⚠ Entire page /auth/amplify/login deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/amplify/login
   Generating static pages (13/374) [==  ] 
 ⚠ Entire page /auth/firebase/login deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/firebase/login
 ⚠ Entire page /auth-demo/modern/login deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth-demo/modern/login
 ⚠ Entire page /auth/firebase/forgot-password deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/firebase/forgot-password
 ⚠ Entire page /auth-demo/classic/forgot-password deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth-demo/classic/forgot-password
 ⚠ Entire page /auth/auth0/login deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/auth0/login
 ⚠ Entire page /auth/firebase/verify deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/firebase/verify
 ⚠ Entire page /auth/supabase/forgot-password deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/supabase/forgot-password
 ⚠ Entire page /auth/amplify/forgot-password deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/amplify/forgot-password
 ⚠ Entire page /auth/firebase/register deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/firebase/register
 ⚠ Entire page /auth-demo/modern/new-password deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth-demo/modern/new-password
 ⚠ Entire page /auth/supabase/login deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/supabase/login
 ⚠ Entire page /components/extra/copy-to-clipboard deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /components/extra/copy-to-clipboard
 ⚠ Entire page /auth/amplify/verify deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/amplify/verify
 ⚠ Entire page /auth/supabase/register deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/supabase/register
 ⚠ Entire page /auth/supabase/new-password deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/supabase/new-password
 ⚠ Entire page /components/extra/image deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /components/extra/image
 ⚠ Entire page /auth/supabase/verify deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth/supabase/verify
 ⚠ Entire page /auth-demo/classic/register deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /auth-demo/classic/register
```


According to the documentation : https://nextjs.org/docs/messages/deopted-into-client-rendering. We need to wrap it in `<Suspense/>`


```js
export default function ProgressBar() {
  return (
    <>
      <NextTopLoader />

      <Suspense fallback={null}>
        <FinishingLoader />
      </Suspense>
    </>
  );
}
```

```js
function FinishingLoader() {
  const pathname = usePathname();

  const router = useRouter();

  const searchParams = useSearchParams();

  useEffect(() => {
    NProgress.done();
  }, [pathname, router, searchParams]);

  return null;
}

```

Reply:
That is because your page is static, i don't have any static page in my app.

Also, please a PR is not a place for debugging, maybe an issue or discussion.